presto-product-tests-launcher/src/main/java/io/prestosql/tests/product/launcher/env/DockerContainer.java

```diff
@@ -312,6 +313,8 @@ public void tryStop()
         catch (Exception e) {
             log.warn("Could not stop container correctly: %s", getStackTraceAsString(e));
         }
+
```

Comment:
You are runnint `tryStop` as:
```java
        // Allow containers to take up to 5 minutes to stop
        Timeout<Object> timeout = Timeout.of(ofMinutes(5))
                .withCancel(true);

        RetryPolicy retry = new RetryPolicy()
                .withMaxAttempts(3);

        FailsafeExecutor<Object> executor = Failsafe.with(timeout, retry);
```
So there is max retries == 3 and timout.
Will the retries be made every timeout/maxretrycount seconds?

Reply:
If above is the case we will almost always wait almost 2 minutes for containers to stop as threre is no significant time window between calling `stop()` in `tryStop()`, and checking if container actually stopped with `isRunning()`